### PR TITLE
[Merged by Bors] - feat: port RingTheory.WittVector.Truncated

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/RingTheory/WittVector/Truncated.lean [hunk 1]
```diff
@@ -213,17 +212,26 @@ instance hasNatPow : Pow (TruncatedWittVector p n R) ℕ :=
...
 
 end TruncatedWittVector
 
-/- ./././Mathport/Syntax/Translate/Expr.lean:336:4: warning: unsupported (TODO): `[tacs] -/
-/-- A macro tactic used to prove that `truncate_fun` respects ring operations. -/
-unsafe def tactic.interactive.witt_truncate_fun_tac : tactic Unit :=
-  sorry
-#align tactic.interactive.witt_truncate_fun_tac tactic.interactive.witt_truncate_fun_tac
+/-- A macro tactic used to prove that `truncateFun` respects ring operations. -/
+macro (name := wtf_tac) "wtf_tac" : tactic =>
```
Comment (jcommelin) [inline Mathlib/RingTheory/WittVector/Truncated.lean:222]: Please rename this to `witt_truncateFun_tac`.

Comment (Parcly-Taxel) [reply Mathlib/RingTheory/WittVector/Truncated.lean]: Done.

File: Mathlib/RingTheory/WittVector/Truncated.lean [hunk 2]
```diff
@@ -447,42 +455,51 @@ variable (f : ∀ k : ℕ, S →+* TruncatedWittVector p k R)
...
 
-`lift` defines the universal property of `𝕎 R` as the inverse limit of `truncated_witt_vector n`.
+`lift` defines the universal property of `𝕎 R` as the inverse limit of `TruncatedWittVector n`.
 -/
-def lift : S →+* 𝕎 R := by
-  refine_struct { toFun := lift_fun f } <;>
-    · intros
-      rw [← sub_eq_zero, ← Ideal.mem_bot, ← infi_ker_truncate, Ideal.mem_iInf]
+def lift : S →+* 𝕎 R := -- Porting note: used `refine_struct`
+  { toFun := liftFun f
```
Comment (semorrison) [inline Mathlib/RingTheory/WittVector/Truncated.lean:486]: We can still use the 
```
refine
  { blah := blah
    .. } <;>
  tac
```
model, and it looks like that would suffice here.

**Comment by jcommelin** [comment]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Pull request closed without merging**